Makefile

```diff
@@ -1,152 +1,245 @@
-		tr -d ':' | \
-		grep -v 'fake' | \
-		tr '\n' '|' | \
-		awk 'BEGIN{print "\\*\\*`("}{print}END{print ")"}' | \
-		tr -d '\n' | \
-		grep -E -f - README.md | \
-		sed 's/\* \*\*`/\o033[32m/' | \
-		sed 's/`\*\*/\o033[0m*/' | \
-		sort | \
-		column -N "Target,Description" -t -s "*"
+	@cat Makefile ${INCLUDE_MAKEFILES} | \
+	   grep -E '^[a-zA-Z_-]+:.*?#' | \
+	   sed 's/:.*# /,/' | \
+	   sort | \
+	   sed 's/^/\o033[32m/' | # Start Green color on first column \
+	   sed 's/,/\o033[0m,/' | # End Green color on first colum \
+	   column -N "Target,Description" -t -s ","
 	@echo ""
 	@echo "Example Usage"
 	@echo "make <target>"
 	@echo "make clear-cache"
 
-init: salt composer-install docker-start ready init-drupal docker-status
+include ${INCLUDE_MAKEFILES}
+
+## removes "No rule to make target" message which allows us to pass an argument
+## without having to specify the name when running the make command
+## https://stackoverflow.com/questions/6273608/how-to-pass-argument-to-makefile-from-command-line/6273809
+##
+## Please see the drush target below for an example
+%:
+	@:
 
-init-drupal: drupal-install config-init config-import clear-cache
 
-update: docker-stop composer-install docker-rebuild ready config-import clear-cache
 
-safe-update: docker-stop composer-install docker-rebuild ready clear-cache
 
-docker-rebuild:
+init: docker-start wait-healthy composer-install init-drupal docker-status # Build environment
+
+safe-update: docker-stop docker-rebuild wait-healthy clear-cache # update without importing config
+
+docker-running:
+	@docker inspect -f '{{.State.Running}}' ${PROJECT}-{db,php,web} &>/dev/null
+
+wait-healthy:
+	@echo "Wait for all containers to become healthy"
+	@python $(CURDIR)/bin/docker-compose-wait.py
+
+docker-rebuild: docker-stop # Update docker images if there has been changes to Dockerfiles
 	docker-compose -f ${DOCKER_COMPOSE_FILE} up -d --build
 	docker-compose -f ${DOCKER_COMPOSE_FILE} ps
-	@sleep 10
 
-docker-status:
+status: docker-status # Alias to docker-status
+docker-status: # Display status of containers related to this project
 	docker-compose -f ${DOCKER_COMPOSE_FILE} ps
 
-docker-start:
+start: docker-start # Alias to docker-start
+docker-start: # Start containers for this project
 	docker-compose -f ${DOCKER_COMPOSE_FILE} up -d
 	docker-compose -f ${DOCKER_COMPOSE_FILE} ps
-	@sleep 10
 
-docker-stop:
+stop: docker-stop # Alias to docker-stop
+docker-stop: # Stop containers for this project
 	docker-compose -f ${DOCKER_COMPOSE_FILE} down
 
-docker-restart: docker-stop docker-start
+docker-restart: docker-stop docker-start # Restart containers for this project
 
-composer-install:
-	composer install --ignore-platform-reqs --no-interaction --no-progress
+drush: # Forwards to drush inside php container
+	docker exec -i ${PROJECT}-php drush $(filter-out $@,$(MAKECMDGOALS))
 
-composer-update:
-	composer update --ignore-platform-reqs --no-interaction --no-progress --prefer-dist
+composer: # Runs composer
```

Comment:
- create /bin/composer that effectively runs `docker run composer` and takes input
- change composer-* targets to use /bin/composer similar to commands that use /bin/drush
- (optional) remove composer installation from php container because reasons